plugins/outputs/bigquery/bigquery.go

```diff
@@ -0,0 +1,215 @@
+package bigquery
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/influxdata/telegraf"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+)
+
+const (
+	testingHostEnv     = "BIGQUERY_TESTING_HOST"
+	defaultOffSetKey   = "offset-key.json"
+	timeStampFieldName = "timestamp"
+)
+
+type BigQuery struct {
+	CredentialsFile string `toml:"credentials_file"`
+	Project         string `toml:"project"`
+	Dataset         string `toml:"dataset"`
+
+	client *bigquery.Client
+}
+
+var sampleConfig = `	
+  ## Credentials File
+  credentials_file = "/path/to/service/account/key.json"
+  ## GCP Project
+  project = "my-gcp-project"
+
+  ## The namespace for the metric descriptor
+  dataset = "telegraf"
+`
+
+func (b *BigQuery) Connect() error {
+	if b.Project == "" {
+		return fmt.Errorf("Project is a required field for BigQuery output")
+	}
+
+	if b.Dataset == "" {
+		return fmt.Errorf("Dataset is a required field for BigQuery output")
+	}
+
+	b.setUpClient()
```

Comment:
For testing you could do
```suggestion
	// To allow testing we do not override the client. The client will never be initialized in
	// normal environment. 
	if b.client == nil {
		b.setUpClient()
	}
```
and init `b.Client` with what you have in `setUpTestClient()`. This way you can move all